src/osd/OpRequest.cc

```diff
@@ -30,12 +30,15 @@ using std::stringstream;
 
 using ceph::Formatter;
 
-OpRequest::OpRequest(Message *req, OpTracker *tracker) :
-  TrackedOp(tracker, req->get_recv_stamp()),
-  rmw_flags(0), request(req),
-  hit_flag_points(0), latest_flag_point(0),
-  hitset_inserted(false)
-{
+OpRequest::OpRequest(Message* req, OpTracker* tracker)
+    : TrackedOp(tracker, req->get_recv_stamp() > req->get_throttle_stamp()
```

Comment:
if the throttle step is always before header_read, you could always use the throttle_stamp here and eliminate the conditional

Reply:
agreed! I'll test it once more to reconfirm and update commit.

Reply:
Yep, can verify the results are satisfactory on replacing `req->get_recv_stamp())` with `req->get_throttle_stamp()`
```
{
                        "event": "initiated",
                        "time": "2019-07-02T12:23:58.513422-0400",
                        "duration": 0
                    },
                    {
                        "event": "throttled",
                        "time": "2019-07-02T12:23:58.513422-0400",
                        "duration": 0.0080613360000000005
                    },
                    {
                        "event": "header_read",
                        "time": "2019-07-02T12:23:58.521483-0400",
                        "duration": 3.5200000000000002e-05
                    },
```